### PR TITLE
expose low-level envoy gRPC to JSON transcoding filter

### PR DESCRIPTION
**Pull request by kdorosh**
# Description

Expose the raw envoy configuration for the gRPC to JSON transcoding filter, which can be leveraged to expose a gRPC service both as a gRPC service and as a REST API. Exposing the underlying envoy configuration allows users more granular control over the gRPC to JSON mappings than the current Gloo API for gRPC to JSON (that doesn't require explicit protobuf descriptors to be provided since they will be discovered). One example where users may want more granular control of their gRPC to JSON mappings may be to leverage query parameter transcoding.

# Context

Labeled WIP until regression test is written, otherwise good for review 👍 

update: test is written, ready for review

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo [contribution guide](https://docs.solo.io/gloo/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2188

### CODE REVIEW

File: projects/gloo/api/v1/options/grpc_json/grpc_json.proto [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+option (extproto.hash_all) = true;
+
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.grpc_json_transcoder.v3";
+option java_outer_classname = "TranscoderProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
```
Comment (yuval-k) [inline projects/gloo/api/v1/options/grpc_json/grpc_json.proto:20]: this proto is owned by us, right?
we probably dont need this nor the java annotations above.
can also remove udpa imports and annotations

File: projects/gloo/pkg/plugins/grpcjson/plugin_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+var _ = Describe("GrpcJson", func() {
+
+	var (
+		initParams     plugins.InitParams
+		expectedFilter []plugins.StagedHttpFilter
+	)
+
+	envoyGrpcJsonConf := &envoy_extensions_filters_http_grpc_json_transcoder_v3.GrpcJsonTranscoder{
+		DescriptorSet: &envoy_extensions_filters_http_grpc_json_transcoder_v3.GrpcJsonTranscoder_ProtoDescriptor{ProtoDescriptor: "/path/to/file"},
```
Comment (yuval-k) [inline projects/gloo/pkg/plugins/grpcjson/plugin_test.go:25]: is this file present for envoy?

File: projects/gloo/pkg/plugins/grpcjson/plugin.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	envoy_extensions_filters_http_grpc_json_transcoder_v3 "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/grpc_json_transcoder/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/wellknown"
+	"github.com/rotisserie/eris"
+	v1 "github.com/solo-io/gloo/projects/gloo/pkg/api/v1"
+	"github.com/solo-io/gloo/projects/gloo/pkg/api/v1/options/grpc_json"
+	"github.com/solo-io/gloo/projects/gloo/pkg/plugins"
+)
+
+// filter info
+var pluginStage = plugins.AfterStage(plugins.AuthZStage)
```
Comment (yuval-k) [inline projects/gloo/pkg/plugins/grpcjson/plugin.go:15]: should we do this During(AcceptedStage)
so that rate limits happen before this filter?

File: projects/gloo/api/v1/options/grpc_json/grpc_json.proto [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+        // Supplies the filename of
+        // :ref:`the proto descriptor set <config_grpc_json_generate_proto_descriptor_set>` for the gRPC
+        // services.
+        string proto_descriptor = 1;
+
+        // Supplies the binary content of
+        // :ref:`the proto descriptor set <config_grpc_json_generate_proto_descriptor_set>` for the gRPC
+        // services.
+        // Note: this must be a base64 standard encoded string; yaml can't handle binary bytes
+        string proto_descriptor_bin = 4;
```
Comment (yuval-k) [inline projects/gloo/api/v1/options/grpc_json/grpc_json.proto:68]: IIRC if you set this to bytes, and set base64 in the yaml, it should decode autmatically

File: projects/gloo/pkg/plugins/grpcjson/plugin_test.go (hunk 2 above)
Comment (kdorosh) [reply projects/gloo/pkg/plugins/grpcjson/plugin_test.go:25]: no, this is test code -- just asserting that the config got propagated (no need for the file to exist for sake of the test)

File: projects/gloo/pkg/plugins/grpcjson/plugin.go (hunk 3 above)
Comment (kdorosh) [reply projects/gloo/pkg/plugins/grpcjson/plugin.go]: https://github.com/solo-io/gloo/pull/3519/commits/7e568a53b21c40e879bb896eacded400f5c7bce9

File: projects/gloo/api/v1/options/grpc_json/grpc_json.proto (hunk 4 above)
Comment (kdorosh) [reply projects/gloo/api/v1/options/grpc_json/grpc_json.proto]: https://github.com/solo-io/gloo/pull/3519/commits/5864b093188c59da2998e87537676b13b8a9dfb0

File: projects/gloo/api/v1/options/grpc_json/grpc_json.proto (hunk 1 above)
Comment (kdorosh) [reply projects/gloo/api/v1/options/grpc_json/grpc_json.proto]: https://github.com/solo-io/gloo/pull/3519/commits/5864b093188c59da2998e87537676b13b8a9dfb0

File: projects/gloo/pkg/plugins/rest/rest.go [hunk 5]
```diff
@@ -105,9 +105,6 @@ func (p *plugin) ProcessRoute(params plugins.RouteParams, in *v1.Route, out *env
 		if err != nil {
 			return nil, err
 		}
-		// should be aws upstream
```
Comment (yuval-k) [inline projects/gloo/pkg/plugins/rest/rest.go:108]: why was this changed?

Comment (kdorosh) [reply projects/gloo/pkg/plugins/rest/rest.go:108]: This logic was duplicated for no reason both [here](https://github.com/solo-io/gloo/blob/v1.5.0-beta18/projects/gloo/pkg/plugins/rest/rest.go#L110) and following eight lines later [here](https://github.com/solo-io/gloo/blob/v1.5.0-beta18/projects/gloo/pkg/plugins/rest/rest.go#L121)
